R/estimator.R

```diff
@@ -35,50 +35,19 @@
 #' to start training.
 #' @param script_params A named list of the command-line arguments to pass to
 #' the training script specified in `entry_script`.
-#' @param cran_packages A list of `cran_package` objects to be installed.
-#' @param github_packages A character vector of GitHub packages to be installed.
-#' @param custom_url_packages A character vector of packages to be installed
-#' from local directory or custom URL.
-#' @param custom_docker_image A string of the name of the Docker image from
-#' which the image to use for training will be built. If not set, a predefined
-#' image will be used as the base image. To use an image from a
-#' private Docker repository, you will also have to specify the
-#' `image_registry_details` parameter.
-#' @param image_registry_details A `ContainerRegistry` object of the details of
-#' the Docker image registry for the custom Docker image.
-#' @param use_gpu Indicates whether the environment to run the experiment should
-#' support GPUs. If `TRUE`, a predefined GPU-based Docker image will be used in the
-#' environment. If `FALSE`, a predefined CPU-based image will be used. Predefined
-#' Docker images (CPU or GPU) will only be used if the `custom_docker_image` parameter
-#' is not set.
-#' @param environment_variables A named list of environment variables names
-#' and values. These environment variables are set on the process where the user
-#' script is being executed.
-#' @param shm_size A string for the size of the Docker container's shared
-#' memory block. For more information, see
-#' [Docker run reference](https://docs.docker.com/engine/reference/run/).
-#' If not set, a default value of `'2g'` is used.
 #' @param max_run_duration_seconds An integer of the maximum allowed time for
 #' the run. Azure ML will attempt to automatically cancel the run if it takes
 #' longer than this value.
 #' @param environment The `Environment` object that configures the R
-#' environment where the experiment is executed. This parameter is mutually
-#' exclusive with the other environment-related parameters `custom_docker_image`
-#' , `image_registry_details`, `use_gpu`, `environment_variables`, `shm_size`,
-#' `cran_packages`, `github_packages`, and `custom_url_packages` and if set
-#' will take precedence over those parameters.
+#' environment where the experiment is executed.
 #' @param inputs A list of DataReference objects or DatasetConsumptionConfig
 #' objects to use as input.
 #' @return The `Estimator` object.
 #' @export
 #' @section Examples:
 #' ```
-#' pkg1 <- cran_package("ggplot2", version = "3.3.0")
```

Comment:
Can we add more detail to the new example so that users don't have to look through the vignettes to see how they'd config an estimator? 

e.g.

``` r
r_env <- r_environment(name = "r-env",
                       version = "1",
                       cran_packages = list(cran_package("dplyr"), cran_package("ggplot2")),
                       use_gpu = TRUE)

est <- estimator(tmp_dir_name,
                 compute_target = existing_compute$name,
                 entry_script = script_name, 
                 script_params = list("data_folder" = ds$as_mount()),
                 environment = r_env)
```

Reply:
done.